File: python/941_ValidMountainArray.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3 or arr[0] > arr[1]: return False
        increasing = True
        for i in range(0, len(arr) - 1):
            if arr[i] == arr[i + 1]: return False
            if increasing: 
                if arr[i] > arr[i + 1]: increasing = False
            else: 
                if arr[i] < arr[i + 1]: return False
        return not increasing

File: java/941_ValidMountainArray.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        if (arr.length < 3 || arr[0] > arr[1]) return false;
        boolean increasing = true;
        for (int i = 0; i < arr.length - 1; ++i) {
            if (arr[i] == arr[i + 1]) return false;
            if (increasing) {
                if (arr[i] > arr[i + 1]) increasing = false;
            } else {
                if (arr[i] < arr[i+1]) return false;
            }
        }
        return !increasing;
    }
}
